lws_config.h.in

```diff
@@ -61,3 +61,5 @@
 
 /* use SHA1() not internal libwebsockets_SHA1 */
 #cmakedefine LWS_SHA1_USE_OPENSSL_NAME
+
+${LWS_SIZEOFPTR_CODE}
```

Comment:
`#cmakedefine LWS_SIZEOFPTR ${LWS_SIZEOFPTR}` would work here, in most cases. Originally, I thought to put it there too but in the end I decided to use just `${LWS_SIZEOFPTR_CODE}` in the lws_config.h.in for the following reasons.

Variable `LWS_SIZEOFPTR` in the CMakeLists.txt gets its value from `CHECK_TYPE_SIZE()` macro, which can [return](https://cmake.org/cmake/help/v3.0/module/CheckTypeSize.html):
* `<size>` - an integer literal if the type size could be determined - for "void *" this would be always true except for certain cases on OS X;
* `"0"` - **a string literal** when building libwebsockets as a universal binary-ready library - cmake will do that on OS X if you use it as e.g., `cmake "-DCMAKE_OSX_ARCHITECTURES=x86_64;i386" <path>`;
* `""` - an empty string - should not be normally possible for "void *".

Thus, in most cases, simply using `#cmakedefine VAR ${VAR}` would result in a `#define VAR <size>` in the lws_config.h file, which would make `_LWS_PAD()` macro in libwebsockets.h work correctly. In case of a multi-architecture build on OS X, though, the build would break completely because there would be `#define VAR "0"` in lws_config.h, which would be painful to filter out with preprocessor directives (if at all possible). It could be analyzed in CMakeLists.txt and substituted with 0 but it would disable the whole feature introduced by commit 173e9c4 for the universal binary use case. So, using `#cmakedefine` would make some users of libwebsockets unhappy if they wanted to build it as a "universal binary"-ready library for their hypothetical multi-architecture OS X application.

The same `CHECK_TYPE_SIZE(type VAR)` expression has on the output actually 3 variables:
* `HAVE_VAR` - if the type exists - true for "void *";
* `VAR` - previously discussed;
* `VAR_CODE` - a set of preprocessor directives to define `VAR` by preprocessor to the correct value, or `/* #undef VAR*/` if the type does not exist (not true in case of "void *").

This last cmake variable expands to a simple `#define VAR <size>` in most mentioned cases, just like `#cmakedefine` does. In case of OS X multi-architecture build, though, it [expands](https://cmake.org/gitweb?p=cmake.git;a=blob;f=Modules/CheckTypeSize.cmake;h=5938d6c9a720dd67124affda8bf6707769d2fb51;hb=HEAD#l256) to a proper set of `#if defined(<arch1>) ... #elif defined(<arch2>) ... #else #error ... #endif` directives, which will work correctly when building the same library for multiple architectures. Thus, it does not introduce a limitation that `#cmakedefine` approach does. *More importantly, since this goes to lws_config.h, it will be included by user code, and will generate correct pointer size for each architecture if that user code happens to be a multi-architecture binary*. Using `#cmakedefine` would put a single number there (if "0" literal is substituted with something sensible).

Using `#cmakedefine VAR ${VAR}` in lws_config.h.in to define the size of "void *" is completely identical to simply using `#define VAR ${VAR}` since that type is a built-in type, and VAR would always be non-empty. `#cmakedefine` is more suited to `HAVE_<feature>` use cases where `<feature>` is much more probable to be absent on a platform.

To summarize, using `${VAR_CODE}` instead of `#cmakedefine VAR ${VAR}` is a compact way of achieving the same goal using the same means but is not redundant and is more portable.

If you still would rather prefer to change that line to `#cmakedefine LWS_SIZEOFPTR ${LWS_SIZEOFPTR}`, I'll do that and will supply a new version of this patch. Thanks.